io_section.tex

```diff
@@ -0,0 +1,257 @@
+\section{I/O}
+The majority of the I/O infrastructure delivering diagnostic output in the form of UGRID-NetCDF via XIOS is on the LFRic trunk
+and will be used as part of the Aquaplanet runs in 2019. Therefore it is now necessary to begin regular monitoring of the impact
+of I/O on compute performance. In 2017, following the preliminary integration of XIOS into the LFRic infrastructure, performance
+tests were done, focussing mainly on strong scaling of LFRic-XIOS for diagnostic output. The conclusion was that XIOS scaled well
+out to 384 nodes (approx. 14K cores). A description of the work plus full results can be found in \cite{Adams2018}. 
+
+For the 2019 report, the focus is on two main areas: strong scaling when outputting regular
+diagnostics and also tests of in-situ post-processing. In particular tests have been done of time-meaning and also regridding 
+from the LFRic native cubed-sphere mesh to a regular lat-lon mesh. Both of these are use cases from the upcoming Aquaplanet
+runs and also evaluation of XIOS regridding is of interest to the wider Next Generation Modelling Systems programme as it
+could inform how downstream processing will be handled. For instance whether LFRic can deliver Level 1 data directly to StaGE.
+It could also make verification of LFRic easier if the data can be directly produced on a lat-lon mesh. The aim of the post-processing
+experiments described here is primarily to get an idea of computational overhead in terms of runtime and memory, 
+so there is no in depth analysis of scientific accuracy apart from verifying that the jobs have produced the right kind of output. 
+
+\subsection{Job Setup}
+All tests were performed on XCS. The executables were built using the LFRic Intel Fortran environment (17.0.0.098) 
+including the latest PSyclone 1.7.0 and XIOS at r1537 of the XIOS trunk. Build flags were the default fast-debug.
+
+The science configuration was the standard baroclinic wave test with timestep length dependent on mesh resolution. 
+For the purposes of the tests here: 900s for a C96 mesh and 180s for a C576 mesh.
+10 fields are currently output as diagnostics in this configuration which results in approximately 5 Gb data per
```

Comment:
Single level fields or multiple levels?

Bigger/dumber question- is it model _levels_ or model _layers_ in the vertical?

Reply:
We call it 'number_of_layers' in the namelist... what exactly that relates to in actual model levels I'm not sure. In the output the fields will be on half or full model levels depending on the field and I guess that is equivalent to UM levels?